src/core/utils/transactions/sender.ts

```diff
@@ -0,0 +1,275 @@
+    const result = await composer.send()
+
+    return this.buildSendAppCallResult(result)
+  }
+
+  async appDelete(params: AppDeleteParams): Promise<SendAppCallResult> {
+    const composer = this.newGroup()
+    composer.addAppDelete(params)
+    const result = await composer.send()
+
+    return this.buildSendAppCallResult(result)
+  }
+
+  private buildSendAppCallMethodCallResult(composerResult: SendTransactionComposerResults): SendAppCallMethodCallResult {
+    return {
+      transaction: composerResult.transactions.at(-1)!,
+      confirmation: composerResult.confirmations.at(-1)!,
+      transactionId: composerResult.transactionIds.at(-1)!,
+      group: composerResult.group,
+      confirmations: composerResult.confirmations,
+      transactionIds: composerResult.transactionIds,
+      transactions: composerResult.transactions,
+      abiReturns: composerResult.abiReturns,
+      abiReturn: composerResult.abiReturns.at(-1),
+    }
+  }
+
+  async appCallMethodCall(params: AppCallMethodCallParams): Promise<SendAppCallMethodCallResult> {
+    const composer = this.newGroup()
+    composer.addAppCallMethodCall(params)
+    const result = await composer.send()
+
+    return this.buildSendAppCallMethodCallResult(result)
+  }
+
+  async appCreateMethodCall(params: AppCreateMethodCallParams): Promise<SendAppCreateMethodCallResult> {
+    const composer = this.newGroup()
+    composer.addAppCreateMethodCall(params)
+    const result = await composer.send()
+
+    const baseResult = this.buildSendAppCallMethodCallResult(result)
+    const applicationIndex = baseResult.confirmation.applicationIndex
+    if (applicationIndex === undefined || applicationIndex <= 0) {
+      throw new Error('App creation confirmation missing applicationIndex')
+    }
+
+    return {
+      ...baseResult,
+      appId: applicationIndex,
+    }
+  }
+
+  async appUpdateMethodCall(params: AppUpdateMethodCallParams): Promise<SendAppCallMethodCallResult> {
+    const composer = this.newGroup()
+    composer.addAppUpdateMethodCall(params)
+    const result = await composer.send()
+
+    return this.buildSendAppCallMethodCallResult(result)
+  }
+
+  async appDeleteMethodCall(params: AppDeleteMethodCallParams): Promise<SendAppCallMethodCallResult> {
+    const composer = this.newGroup()
+    composer.addAppDeleteMethodCall(params)
+    const result = await composer.send()
+
+    return this.buildSendAppCallMethodCallResult(result)
+  }
+
+  async onlineKeyRegistration(params: OnlineKeyRegistrationParams): Promise<SendResult> {
+    const composer = this.newGroup()
+    composer.addOnlineKeyRegistration(params)
+    const result = await composer.send()
+
+    return this.buildSendResult(result)
+  }
+
+  async offlineKeyRegistration(params: OfflineKeyRegistrationParams): Promise<SendResult> {
+    const composer = this.newGroup()
+    composer.addOfflineKeyRegistration(params)
+    const result = await composer.send()
+
+    return this.buildSendResult(result)
+  }
```

Comment:
Looks like this abstraction is missing the following:
- `accountClose`
- `addAssetClawback`
- `assetUnFreeze`
- `nonParticipationKeyRegistration`